apps/aecore/lib/aecore/chain/block_validation.ex

```diff
@@ -176,4 +179,28 @@ defmodule Aecore.Chain.BlockValidation do
   defp check_correct_height?(new_block, previous_block) do
     previous_block.header.height + 1 == new_block.header.height
   end
+
+  @spec valid_header_timestamp(Block.t()) :: boolean()
+  defp valid_header_timestamp(%Block{header: new_block_header}) do
+    case new_block_header.timestamp <= :os.system_time() + 3_600_000 do
+      true ->
+        top_block = Chain.top_block()
+        last_blocks = Chain.get_blocks(top_block, 10)
+
+        last_blocks_timestamps =
+          Enum.reduce(last_blocks, [], fn block, acc -> acc = acc ++ [block.header.timestamp] end)
+
+        average = Enum.sum(last_blocks) / 10
+
+        if new_block_header.timestamp > average do
```

Comment:
`new_block_header.timestamp > average` returns true or false on its own so this if isn't needed